src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -1102,3 +1104,279 @@ class FiveLMMountConfig:
+
+
+"""
+Utilities for generating multi-LM dataset args.
+"""
+
+
+def make_sensor_positions_on_grid(
+    n_sensors: int,
+    delta: Number = 0.01,
+    order_by: str = "distance",
+    add_view_finder: bool = True,
+) -> np.ndarray:
+    """Generate sensor positions on a 2D grid.
+
+    Create mounting positions for an arbitrary number of sensors, where the
+    sensors lie on an imaginary grid on the xy plane (and z = 0). Sensor position
+    0 is always centered at (0, 0, 0), and all other sensors are clustered
+    around it. The method for selecting which grid points around the center to assign
+    each sensor is determined by the `order_by` argument (see below).
+
+    By default, `n_sensors + 1` positions are returned; the first `n_sensors` positions
+    are for regular sensors, and an additional position is appended by default to
+    accommodate a view finder. The view finder position (if used) is the same as
+    sensor position 0 (i.e., (0, 0, 0)).
+
+    Args:
+        n_sensors (int): Number of sensors. Count should not include a view finder.
+        delta (number): The grid spacing length. By default, sensors will be
+            placed every centimeter (units are in meters).
+        order_by (str, optional): How to select points on the grid that will contain
+            sensors.
+             - "spiral": sensors are numbered along a counter-clockwise spiral
+                spreading outwards from the center.
+             - "distance": sensors are ordered by their distance from the center.
+                This can result in a more jagged pattern along the edges but
+                results in sensors generally more packed towards the center.
+                Positions that are equidistant from the center are ordered
+                counterclockwise starting at 3 o'clock.
+        add_view_finder (bool, optional): Whether to include an extra position module
+            at the origin to serve as a view finder. Defaults to `True`.
+
+    Returns:
+        np.ndarray: A 2D array of sensor positions where each row has positions
+            (x, y, z). If `add_view_finder` is True, the array has `n_sensors + 1` rows,
+            where the last row corresponds to the view finder's position and is
+            identical to row 0. Otherwise, the array has `n_sensors` rows. row 0 is
+            always centered at (0, 0, 0), and all other rows are offset relative to it.
+
+    """
+    assert n_sensors > 0, "n_sensors must be greater than 0"
+    assert delta > 0, "delta must be greater than 0"
+    assert order_by in ["spiral", "distance"], "order_by must be 'spiral' or 'distance'"
+
+    # Find smallest square grid size that can fit n_lms with odd-length sides.
+    grid_size = 1
+    while n_sensors > grid_size**2:
+        grid_size += 2
+
+    # Make coordinate grids, where the center is (0, 0).
+    points = np.arange(-grid_size // 2 + 1, grid_size // 2 + 1)
+    x, y = np.meshgrid(points, points)
+    y = np.flipud(y)
```

Comment:
What does this line do `y = np.flipud(y)`? Sorry just struggling to visualize it.

Reply:
`meshgrid` orders y-values from smallest (and negative) to largest when going top-to-bottom in the matrix. I flipped it so "up" is the positive y-direction to match habitat's coordinate system.

Reply:
Ah nice thanks, maybe worth adding a short comment for that.